lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -93,6 +95,12 @@ defmodule Mix.Tasks.Deps.Compile do
     if true in compiled, do: Mix.Dep.Lock.touch_manifest, else: :ok
   end
 
+  defp clean(dep, opts) do
+    if Keyword.get(opts, :force, false) do
+      Mix.Tasks.Deps.Clean.run [dep, "--build"]
```

Comment:
I think this could simply be:

    File.rm_rf! Path.join [Mix.Project.build_path, "lib", Atom.to_string(app)]

What do you think? Also, let's call the argument `app` instead of `dep`. :)